smurfs/src/actions/actions.js

```diff
@@ -0,0 +1,50 @@
+import axios from 'axios';
+//action types
+export const FETCH_SMURFS_START = 'FETCH_SMURFS_START';
+export const FETCH_SMURFS_SUCCESS = 'FETCH_SMURFS_SUCCESS';
+export const FETCH_SMURFS_FAILURE = 'FETCH_SMURFS_FAILURE';
+export const ADD_SMURFS = 'ADD_SMURFS';
+export const REMOVE_SMURFS = 'REMOVE_SMURFS';
+
+//action creator
+
+export const getSmurf = () => dispatch => {
+    dispatch({ type: FETCH_SMURFS_START });
+    axios.get('http://localhost:3333/smurfs')
+        .then(res => {
+            dispatch({ type: FETCH_SMURFS_SUCCESS, payload: res.data });
+        })
+        .catch(err => {
+            dispatch({ type: FETCH_SMURFS_FAILURE, payload: err.response })
+        })
+};
+
+export const addSmurf = smurf => dispatch => {
+    dispatch({ type: ADD_SMURFS });
+    axios.post('http://localhost:3333/smurfs', smurf)
+        .then(res => {
+            console.log(res.data)
+            dispatch({ type: ADD_SMURFS, payload: res.data })
+            dispatch({ type: FETCH_SMURFS_SUCCESS, payload: res.data })
+        })
+
+        .catch(err => {
+            console.log(err);
+        });
+}
+
+// export const removeSmurfs = smurf => dispatch => {
+//     return { type: 'REMOVE_SMURFS', payload: smurf }
+// }
+
+// export const  removeSmurf=(smurf) => dispatch =>{
+//     dispatch({type: REMOVE_SMURFS});
+//       axios.delete('/smurfs').then(() => {
+//         console.log(`Deleted ${smurf.id}`)
+
```

Comment:
Awesome that you tried doing this stretch, seriously. Consider this: You want to remove one of the smurfs from the list of smurfs you have stored in your state, right? That state is an array of smurfs...what would be the best way to remove an item from an array and return that new array without that specific item in it anymore? I think you have the right idea because i can see in your reducer you were trying to use filter...its **very** close!
Go back over the ReadMe where it talks about the stretch for deleting. 
When it says DELETE '/smurfs/123', where 123 is the Id of the smurf you want to remove that /smurfs/123 - all they are asking is that you tag the id of the smurf you want to delete to the end of the endpoint....remember that the endpoint. what endpoint have you been using to GET the smurfs? All you need to do for the DELETE is to put /id-of-smurf at the end. You will probably want to use an interpolated string kinda like how you used it on line 43.